src/Microsoft.VisualStudio.Editor.Razor/DefaultTemplateEngineFactoryService.cs

```diff
@@ -41,11 +42,13 @@ public override RazorTemplateEngine Create(string projectPath, Action<IRazorEngi
             // In 15.5 we expect projectPath to be a directory, NOT the path to the csproj.
             var project = FindProject(projectPath);
             var configuration = (project?.Configuration as MvcExtensibilityConfiguration) ?? DefaultConfiguration;
+            var razorLanguageVersion = configuration.RazorLanguageVersion;
+            var razorConfiguration = new RazorConfiguration(razorLanguageVersion, designTime: true);
 
             RazorEngine engine;
-            if (configuration.RazorAssembly.Identity.Version.Major == 1)
+            if (razorLanguageVersion.Major == 1)
             {
-                engine = RazorEngine.CreateDesignTime(b =>
+                engine = RazorEngine.CreateCore(razorConfiguration, b =>
```

Comment:
Looks good. I assume this is the meat of it. The pieces come together pretty well.